app/controllers/admin/student_accounts_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Admin
+    class StudentAccountsController < BaseController
+      layout 'admin'
+  
+      def actions
+      end
+  
+      def get_number_of_accounts
+        year = get_year
+        num_of_accounts = query(year)
+        flash[:notice] = "The number of student acccounts created since July 1 is #{num_of_accounts}"
+        redirect_to actions_admin_student_accounts_path
+      end
+
+      def get_year
+        # correct year for July 1
+        today = DateTime.now
+        year = today.year
+        month = today.month
+        if month < 7
+            year = year - 1
+        end
+        return year
+      end
+
+      def query(year)
+        role_type = :student
+        user_ids = User.where(role: role_type)
+        return SecurityLog.where(event_type: :sign_up_successful, user_id: user_ids).where('created_at >= ?', Date.parse(year.inspect + '-07-01')).count
```

Comment:
@BryanEliDimas can check me on this, but I _think_ that a created `User` record is a valid, succesfully-signed-up user (i.e. that we don't have users in incomplete states that never fully signed up).

If as discussed in the other comment we switch to putting this query in the user reports page, we should then able to follow the [pattern of queries there](https://github.com/openstax/accounts/blob/master/app/views/admin/reports/show.html.erb#L5-L17) and compute the count you want with:

```ruby
User.student.where(created_at: DateTime.last_july_1..DateTime.now).count
```

Breaking that down...

`.student` is a utility method provided by the [`enum :role` line](https://github.com/openstax/accounts/blob/master/app/models/user.rb#L37) in the `User` model, it is the same as saying `.where(role: :student)`.  See more about ActiveRecord enums [here](https://api.rubyonrails.org/v5.2.4.1/classes/ActiveRecord/Enum.html)

`DateTime.last_july_1..DateTime.now` returns a [Ruby range](https://ruby-doc.org/core-2.5.1/Range.html) with a start time of last July and an end time of right now.

`.where(created_at: some_range)` does a query where `created_at` falls in that range

Reply:
Done